modules/admin/src/test/java/org/motechproject/admin/it/AdminBundleIT.java

```diff
@@ -114,6 +127,64 @@ public void testMessageController() throws IOException, InterruptedException {
         assertTrue("No messages listed", json.size() > 0);
     }
 
+    @Test
+    public void testUploadBundleFromRepository() throws IOException, InterruptedException {
+        boolean isUploadSuccessful = uploadBundle("Repository", "org.motechproject:cms-lite:LATEST", null,
+                "on","org.motechproject.cms-lite");
+
+        assertTrue(isUploadSuccessful);
+    }
+
+    @Test
+    public void testUploadBundleFromFile() throws IOException, InterruptedException {
+        File file = new File("src/test/resources/motech-upload-test-bundle.jar");
+
+        boolean isUploadSuccessful = uploadBundle("File", null, file, "on", "motech-upload-test-bundle");
+
+        assertTrue(isUploadSuccessful);
+    }
+
+    private boolean uploadBundle(String moduleSource, String moduleId, File bundleFile,
+                                 String startBundle, String bundleSymbolicName) throws IOException, InterruptedException {
+        String uri = String.format("http://%s:%d/admin/api/bundles/upload/", HOST, PORT);
+
+        HttpPost httpPost = new HttpPost(uri);
+        MultipartEntityBuilder entity = MultipartEntityBuilder.create();
+        Charset chars = Charset.forName("UTF-8");
+        entity.setCharset(chars);
+        entity.addTextBody("moduleSource", moduleSource, ContentType.MULTIPART_FORM_DATA);
+        if (moduleSource.equals("Repository")) {
+            entity.addTextBody("moduleId", moduleId, ContentType.MULTIPART_FORM_DATA);
+        }
+        else if (moduleSource.equals("File")) {
+            if (bundleFile == null) {
+                fail("Bundle file was null.");
+            }
+            entity.addBinaryBody("bundleFile", bundleFile);
+        }
+        else {
+            fail("Wrong module source.");
+        }
+        entity.addTextBody("startBundle", startBundle, ContentType.MULTIPART_FORM_DATA);
+        httpPost.setEntity(entity.build());
+
+        int bundlesCountBeforeUpload = bundleContext.getBundles().length;
+        HttpResponse response = getHttpClient().execute(httpPost);
+        EntityUtils.consume(response.getEntity());
+        int bundlesCountAfterUpload = bundleContext.getBundles().length;
+
+        if (response.getStatusLine().getStatusCode() != HttpStatus.ORDINAL_200_OK) {
+            fail("Wrong response code.");
+        }
+
+        Bundle uploadedBundle = getBundleFromBundlesArray(bundleSymbolicName);
+        if(uploadedBundle == null) {
+            fail("Couldn't find uploaded bundle in current bundles.");
+        }
+
+        return (bundlesCountAfterUpload == bundlesCountBeforeUpload + 1) && (uploadedBundle.getState() == Bundle.ACTIVE);
```

Comment:
Those two checks should also use assertEquals, rather than returning booleans. This method should probably be void, we do not need to return anything.